### PR TITLE
Have the position restraint deals with the alchemical ion

### PR DESCRIPTION
**Pull request by xiki-tempula**

### CODE REVIEW

**Comment by xiki-tempula** [comment]
Not ready for review yet.

File: python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py [hunk 1]
```diff
@@ -304,23 +310,32 @@ def generateAmberConfig(self, extra_options=None, extra_lines=None):
...
         # Pressure control.
         if not isinstance(self.protocol, _Protocol.Minimisation):
             if self.protocol.getPressure() is not None:
                 # Don't use barostat for vacuum simulations.
                 if self._has_box and self._has_water:
                     protocol_dict["ntp"] = 1  # Isotropic pressure scaling.
-                    protocol_dict[
-                        "pres0"
-                    ] = f"{self.protocol.getPressure().bar().value():.5f}"  # Pressure in bar.
+                    protocol_dict["pres0"] = (
```
Comment (xiki-tempula) [inline python/BioSimSpace/Sandpit/Exscientia/Protocol/_config.py:336]: Nothing changed after this line for this file.

File: tests/Sandpit/Exscientia/Process/test_amber.py [hunk 2]
```diff
@@ -5,11 +5,9 @@
 import numpy as np
 import pandas as pd
 import pytest
-import shutil
 
 import BioSimSpace.Sandpit.Exscientia as BSS
-
-from tests.Sandpit.Exscientia.conftest import url, has_amber, has_pyarrow
+from tests.Sandpit.Exscientia.conftest import has_amber, has_pyarrow
```
Comment (xiki-tempula) [inline tests/Sandpit/Exscientia/Process/test_amber.py:10]: This file not changed.

File: tests/Sandpit/Exscientia/Process/test_gromacs.py [hunk 3]
```diff
@@ -1,13 +1,12 @@
 import math
-import numpy as np
-import pytest
-import pandas as pd
 import shutil
-
 from pathlib import Path
 
-import BioSimSpace.Sandpit.Exscientia as BSS
+import numpy as np
```
Comment (xiki-tempula) [inline tests/Sandpit/Exscientia/Process/test_gromacs.py:5]: This file not changed.

File: python/BioSimSpace/Sandpit/Exscientia/Align/_alch_ion.py [hunk 4]
```diff
@@ -50,3 +50,23 @@ def _mark_alchemical_ion(molecule):
     mol._sire_object = mol_edit.commit()
 
     return mol
+
+
+def _get_protein_com_idx(system: _System) -> int:
+    """return the index of the atom that is closet to the center of
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Align/_alch_ion.py]: `Return the index of the atom that is closest`

File: python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py [hunk 5]
```diff
@@ -543,7 +546,10 @@ def _generate_args(self):
         if not isinstance(self._protocol, _Protocol.Custom):
             # Append a reference file if this a restrained simulation.
             if isinstance(self._protocol, _Protocol._PositionRestraintMixin):
-                if self._protocol.getRestraint() is not None:
+                if (
+                    self._protocol.getRestraint() is not None
+                    or self._system.getAlchemicalIon()
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py:551]: Feels like in general we should have the option not to restrain the alch ion as part of the protocol but this is fine as is.

File: python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_molecule.py [hunk 6]
```diff
@@ -1879,6 +1881,20 @@ def _getPerturbationIndices(self):
 
         return idxs
 
+    def getCOMIdx(self):
+        """Get the index of the atom that closest to the center of mass."""
+        if self.isPerturbable():
+            property_map = {"coordinates": "coordinates0", "mass": "mass0"}
+        else:
+            property_map = {"coordinates": "coordinates", "mass": "mass"}
+        coords = self.coordinates(property_map=property_map)
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_molecule.py:1890]: I think there was a `sire` function which gives you the numpy array of coordinates

**Review by msuruzhon — APPROVED** [review]

**Pull request merged**